rest/client/request.go

```diff
@@ -198,6 +200,12 @@ func (c *communicatorImpl) retryRequest(ctx context.Context, info requestInfo, d
 				return resp, nil
 			} else if resp.StatusCode == http.StatusConflict {
 				return nil, HTTPConflictError
+			} else if resp.StatusCode >= 400 && resp.StatusCode < 500 {
+				apiErr := rest.APIError{}
+				if err := util.ReadJSONInto(resp.Body, err); err != nil {
+					return nil, errors.Wrapf(apiErr, "server returned %d", http.StatusBadRequest)
```

Comment:
Wrapf(&apiError), because errors are interfaces. (or change `apiError := rest...` to `apiError := &rest...`)